src/oop/static_modifier/medium/q3/MainTest.java

```diff
@@ -0,0 +1,230 @@
+                String.format("Your %s attribute is missing the static modifier.", stringAttribute1));
+        assertTrue(classInstance.hasModifier(stringAttribute2, "private"),
+                String.format("Your %s attribute does not have the correct visibility modifier.", stringAttribute2));
+    }
+
+    private static Stream<String> constructorInputProvider() {
+        return Stream.of("Mary", "Jane", "Jeremy");
+    }
+
+    @Test
+    public void horseClassHasCorrectConstructor() {
+        Class<?>[] arguments = {String.class};
+        assertTrue(classInstance.hasConstructor(arguments),
+                String.format("Your %s class is missing a required constructor.", objectName));
+        assertTrue(classInstance.hasConstructor(arguments, new String[]{"public"}),
+                String.format("Your %s class constructor has the incorrect visibility modifier.", objectName));
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void horseConstructorInitializesAttributesCorrectly(String n) throws Throwable {
+        Object[][] arguments = {
+                {n, String.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        String incorrectAttributeValue =
+                "Your %s constructor does not correctly initialize the %s attribute based on the passed parameters.";
+        _assertEquals(n, classInstance.getFieldValue(instance, stringAttribute2),
+                String.format(incorrectAttributeValue, objectName, stringAttribute2));
+    }
+
+    @Tag("dependent1")
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void horseClassHasCorrectToStringMethod(String name) throws Throwable {
+        classInstance.setFieldValue(null, startNoise, stringAttribute1);
+        Object[][] arguments = {
+                {name, String.class},
+        };
+        Object instance = classInstance.createInstance(arguments);
+        assertTrue(classInstance.hasMethod("toString", new Class[]{}),
+                "Your " + objectName + " class is missing the toString method.");
+        assertTrue(classInstance.hasMethod("toString", new Class[]{}, String.class),
+                "Your " + objectName + " class toString method does not have the correct return type.");
+        assertTrue(classInstance.hasMethod("toString", new Class[]{}, String.class, new String[]{"public"}),
+                "Your " + objectName + " class toString method does not have the correct visibility modifier.");
+        String expectedOutput = name + " calls out " + startNoise;
+        Object output = classInstance.callMethod("toString", new String[]{"public"}, instance);
+        String incorrectToString = String.format("Your toString method for the %s class does return the correct String.", objectName);
+        _assertEquals(expectedOutput, output, incorrectToString);
+    }
+
+    private static Stream<Arguments> methodInputProvider() {
```

Comment:
Should we also test if there are no vowels in the word? Or does this result in undefined behaviour?